### DISCUSSION TITLE
【IOS】Is there any way i can generate a video which record the filament's view with `opengl` backend?

### DISCUSSION THREAD

**Discussion by dandingol03** [google/filament#5034] [Q&A]
Hi . 
Recently, i decide  to choose `opengl` as engine backend which i am more familiar with. I want to generate a video which record the filament's stuff.
the **process** when i use `metal` as backend before:
1. generate a `CVPIxelBuffer`
2. use the CVPixelBuffer to generate `Swapchain`
3. render frame with swapchain
4. write CVPixelBuffer to `AVAssetInput`

However,  the `CVPixelbuffer` doesn't work on `opengl` backend. How should i change the process to generate video correctly? I come to seek some help, thanks.

**Comment by bejado** [comment]
`engine backend which i am more familiar with`

Our goal is to make Filament's API agnostic to the backend, so it shouldn't matter how familiar you are with the graphics API itself. I'd be curious to hear about any issues you've run into.
